### PR TITLE
Add note in docs about needs_parentheses

### PR DESCRIPTION
**Pull request by thofma**

### CODE REVIEW

File: docs/src/ncrings.md [hunk 1]
```diff
@@ -259,6 +264,18 @@ Should returns `true` if parentheses are needed around this object when printed,
 a coefficient of a polynomial. As an example, non-constant polynomials would need such
 parentheses if used as coefficients of another polynomial.
 
+Note that since this approach quickly leads to unnecessary parentheses, the
+expression method below is preferred.
+
+### Expressions
+
+To obtain best results when printing composed types derived from other types, e.g., polynomials,
+the following method should be implemented
```
Comment (fingolfin) [inline docs/src/ncrings.md:273]:
```suggestion
the following method should be implemented.
```

Comment (fingolfin) [reply docs/src/ncrings.md:273]: This leaves it unclear whether one needs to implement both `show` and `expressify`, and how they relate.

Looking at the AA source code, it seems that if one implements `expressify`, one can implement `show` as
```julia
function Base.show(io::IO, a::SOME_TYPE)
  print(io, AbstractAlgebra.obj_to_string(a, context = io))
end
```
If so, perhaps that could be stated?

File: docs/src/rings.md [hunk 2]
```diff
@@ -335,13 +340,14 @@ expression method below is preferred.
 
 ### Expressions
 
-Alternatively, one can implement a method
+To obtain best results when printing composed types derived from other types, e.g., polynomials,
+the following method should be implemented
```
Comment (fingolfin) [inline docs/src/rings.md:344]: meta-comment: I really dislike that the ncring and ring documentation overlap so much, and duplicate tons of stuff. That just adds work, leads to drift, and also is worse for the user, who now has to go over everything with a comb to find the difference. Why not have a page which lists all the things that are the same; and then ring and ncring just say "everything from that page, plus the following few points" ? Or alternatively, keep the ncring page, and change the ring page to say "this is like ncring, except that the following are added: ... ; and the following are replaced by this and that ..."

Comment (rfourquet) [reply docs/src/rings.md:344]: I couldn't agree more with that.

Comment (thofma) [reply docs/src/rings.md]: I agree, but this is out of scope of this PR. Might be worthwhile to open an issue.

**Comment by thofma** [comment]
Addressed the comments.

File: docs/src/rings.md (hunk 2 above)
Comment (fingolfin) [reply docs/src/rings.md]: Indeed it is outside the scope of this PR (that's what I meant to convey with "meta-comment", I'll make sure to write it explicitly in the future).

I've filed issue #725 about this.

File: docs/src/ncrings.md [hunk 3]
```diff
@@ -259,6 +264,18 @@ Should returns `true` if parentheses are needed around this object when printed,
...
+
+### Expressions
+
+To obtain best results when printing composed types derived from other types, e.g., polynomials,
+the following method should be implemented.
+
+```julia
+expressify(f::MyElem; context = nothing)
+```
+
```
Comment (fingolfin) [inline docs/src/ncrings.md:278]: Perhaps duplicate the rings version of this section here? Or alternatively, add something like
> For more information about `expressify`, please refer to the "Ring Interface" section"

**Review by fingolfin — APPROVED** [review]
Certainly a nice improvement. I have one more minor suggestion, but also OK to merge as-is

File: docs/src/rings.md (hunk 2 above)
Comment (thofma) [reply docs/src/rings.md]: I now what you meant. I just wrote that this is orthogonal to this pull request and that you should open and issue if you don't want your comment to be forgotten. It has nothing to do with it being explicit or not.

**Comment by thofma** [comment]
I addressed the final comment.

**Pull request merged**